### PR TITLE
[AzureMonitorDistro] resync vendored code

### PR DESCRIPTION
**Pull request by TimothyMothra**
Follow up to https://github.com/Azure/azure-sdk-for-net/pull/42479#discussion_r1515439382

Because we've been manually syncing vendored files, some of these files are out of sync with the OpenTelemetry repo. I've resynced all files.

## Explanation of Changes
- removed `// <copyright file="FILENAME" company="OpenTelemetry Authors">` from all vendored files.
  This was a mistake made when manually syncing files.
- revert class name `SqlClient_TracerProviderBuilderExtensions`. The "SqlClient_" prefix was necessary when we had multiple vendored instrumentation libraries with identical class names/namespaces. This is no longer needed.

## Steps followed to Sync Vendored code

1. Create a New Branch in azure-sdk-for-net repo. This is where changes will be committed. 
2. Sync `OpenTelemetry.Instrumentation.SqlClient`
    2a. Clone the OpenTelemetry.NET Repository
    ```bash
    git clone https://github.com/open-telemetry/opentelemetry-dotnet.git
    ```
    2b. Fetch the tags and checkout the most recent release
    ```bash
    git fetch --tags
    git checkout tags/Instrumentation.SqlClient-1.7.0-beta.1
    ```
    2c. Copy the following folders into `Azure.Monitor.OpenTelemetry.AspNetCore`'s `Vendoring` directory:
    - `OpenTelemetry.Instrumentation.SqlClient`
        - Remove the following files:
            - *.csproj
            - README.md
            - CHANGELOG.MD
            - AssemblyInfo.cs
    - `Shared`
        - Specifically copy these files:
            - DiagnosticSourceInstrumentation\DiagnosticSourceListener.cs
            - DiagnosticSourceInstrumentation\DiagnosticSourceSubscriber.cs
            - DiagnosticSourceInstrumentation\ListenerHandler.cs
            - DiagnosticSourceInstrumentation\PropertyFetcher.cs
            - Shims\IsExternalInit.cs
            - Shims\NullableAttributes.cs
            - ExceptionExtensions.cs
            - Guard.cs
            - ResourceSemanticConventions.cs
            - SemanticConventions.cs
            - SpanAttributeConstants.cs
3. Sync `OpenTelemetry.ResourceDetectors.Azure`
    3a. Clone the OpenTelemetry.NET Contrib Repository
    ```bash
    git clone https://github.com/open-telemetry/opentelemetry-dotnet-contrib.git
    ```
    3b. Fetch the tags and checkout the most recent release
    ```bash
    git fetch --tags
    git checkout tags/ResourceDetectors.Azure-1.0.0-beta.5
    ```
    3c. Copy the following folders into the `Azure.Monitor.OpenTelemetry.AspNetCore`'s `Vendoring` directory:
    - `OpenTelemetry.ResourceDetectors.Azure`
        - Remove the following files:
            - *.csproj
            - README.md
            - CHANGELOG.MD
            - AssemblyInfo.cs
    - `Shared`
        - Must do a file comparison to confirm there are no missing changes or conflicts between the two repos.
4. Additional Changes
    - Modified all instrumentation public APIs to internal.
    - Added `#nullable enable` to `OpenTelemetry.ResourceDetoctors.Azure` files. Our `Azure.Monitor.OpenTelemetry.AspNetCore` project does not yet support nullables.
    - Added `#pragma warning disable` `AZC0102` and `AZC0104` to `Azure.VmMetaDataRequestor.cs`. These are code standards that only exist in this Azure SDK reop.

### CODE REVIEW

**Comment by TimothyMothra** [comment]
> > When syncing file updates, there was some confusion on our end about what copyright should be. I re-reviewed our conversation with CELA, and their instruction was to copy the copyright as-is.
> 
> I don't think that's the right way to think about this issue. The right model IMHO is [#42479 (comment)](https://github.com/Azure/azure-sdk-for-net/pull/42479#discussion_r1516626714)

@reyang please re-review. I resynced the files using the model discussed. I've updated this PR and the description.

**Comment by reyang** [comment]
> > > When syncing file updates, there was some confusion on our end about what copyright should be. I re-reviewed our conversation with CELA, and their instruction was to copy the copyright as-is.
> > 
> > 
> > I don't think that's the right way to think about this issue. The right model IMHO is [#42479 (comment)](https://github.com/Azure/azure-sdk-for-net/pull/42479#discussion_r1516626714)
> 
> @reyang please re-review. I resynced the files using the model discussed. I've updated this PR and the description.

The description/approach looks good to me, Thanks! Don't get blocked on me (I don't plan to spend time to compare files and approve the PR).

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM.
Please have Vishwesh review the PR before the merge.

**Review by vishweshbankwar — APPROVED** [review]
LGTM

**Pull request merged**
